src/render.js

```diff
@@ -241,22 +254,30 @@ class App extends Preact.Component {
       },
       () => {
         alert(`Deleted ${length} entries.`);
-      }
+      },
     );
   };
 
-  render(props, state) {
-    const isSelecting =
-      this.state.selecting && this.state.history.some((e) => e.selected);
+  /**
+   * @returns {boolean}
+   */
+  isSelecting = () => {
```

Comment:
It'd be better to keep it as a local variable inside `render` since it's not needed outside of it. No need to keep it alive more than it should! 😄

Reply:
it is needed in case of shortcuts, deleting without selection is useless 😄 
![image](https://user-images.githubusercontent.com/57839948/107555607-a7a9dd00-6bd7-11eb-90e2-f30b097ab949.png)

Reply:
notice: I simulate user click to unfocus the selected entries

Reply:
Oh, I compared against `master`'s code instead of looking at your branch 😂🤦 By all means feel free to leave it as is. 

What do you think about using `App#copySelection` directly instead of going for that loop? I don't think copying using keyboard shortcuts should invalidate previous selections, to keep it seamless with file explorer.

Reply:
Using `App#copySelection` directly, causes the outline stay on selected entries

Reply:
There's a use case for it. As a user:
1. I hold shift
2. Select multiple entries that I wish to copy
3. I then hit `CTRL + C` to copy
4. Oops! I wanted to copy 1 more entry with those
5. Now I have to re-select all of them

Reply:
ok, I will change that

Reply:
but what about `App#deleteSelection`?
it also needs to know if user is selecting

Reply:
What do you mean?

Reply:
`App#deleteSelection` will always try to delete selected entries
![image](https://user-images.githubusercontent.com/57839948/107563760-c01ef500-6be1-11eb-8e49-e5aaed359d76.png)

Reply:
I'm still not on the same page. `deleteSelection` filters the history once invoked, independent of whether you're holding the `Shift`. What's the problem?

Reply:
Are you referring to [this](https://github.com/KL13NT/clippy/blob/09f79d2190791a793424d856a550807b738450e2/src/render.js#L100) condition?

Reply:
I'm saying about this:
https://github.com/KL13NT/clippy/blob/09f79d2190791a793424d856a550807b738450e2/src/render.js#L97

Reply:
Oh, yeah, I meant to refer to that one 😂
Let's see. If you've selected a bunch of items, and you release the `Shift` key then the `state.selecting` will be `false`. So, modifying the `isSelecting` method to yield `state.selecting || some elements are selected` instead of the `&&` would fix this, wouldn't it?

Reply:
1. I think that I broke the keyup listener
2. isSelecting method works fine

Reply:
Let's take a step back and look at it. `App#deleteSelection` doesn't care whether you're holding stuff. It filters the history to remove any entries with `selected=true`. So, in that sense:
1. Hold `Shift`, state is: `{selecting:true}`
2. Select multiple entries, state is: `{selecting:true} with some entries selected=true`
3. Keyup `Shift`, state is: `{selecting: false} with some entries selected=true`
4. `App#copySelection` will turn all entries with state `selected=true` to string and copy them
5. `App#deleteSelection` will delete those entries with state `selected=true` either way so you don't need it to be conditional

Now the condition on line 97 causes the shortcut call to `App#deleteSelection` to depend on having the `Shift` key activated. Modifying `isSelecting` it to use `||` OR instead of `&&` AND will tell the handler "despite the user not holding the Shift key, they have a few entries selected". 

So switching it to `||` *should* fix it. Give it a shot, and tell me if you would like me to debug that while you resolve the other conversations?

Reply:
It seems that release the shift doesn't change the state

Reply:
This would require a ripple-modification of `handleKeyUpClick`.

Reply:
> 
> 
> It seems that release the shift doesn't change the state


Tell me if you would like me to hand this bug to me. I'll gladly work on it.

Reply:
I think this line is wrong:
https://github.com/KL13NT/clippy/blob/09f79d2190791a793424d856a550807b738450e2/src/render.js#L107

Reply:
1. There are some selected entries: does not change the state 
2. There aren't any selected entries: state changes

Reply:
The condition set `state.selecting` to false if you release the shift key and have not selected anything. Not sure how it's wrong.

Reply:
https://github.com/KL13NT/clippy/pull/37#discussion_r574052160

3.Keyup Shift, state is: {selecting: false} with some entries selected=true

Reply:
it will be true, because 
```js 
this.state.history.some((e) => e.selected)
```
will return true, and it will be converted to false so condition is false

Reply:
Oh, that's a sloppy one by me. Should be 3. Keyup Shift, state is: {selecting: true}. I'll edit it now.

Reply:
This issue seems to be taking too much effort. You can carry on using the delete button `ref` as long as it worked, and we can look at this at another issue.

Reply:
I was using only copySelection button for ref

Reply:
I will push the changes now

Reply:
As long as everything works well just move on with it. If I find issues during the merge I'll either raise new bug reports to move this discussion there or fix them immediately. You put a ton of effort into this PR so far that I absolutely admire so let's wrap this PR up and see what happens! 😅